src/auth_v3.js

```diff
@@ -271,10 +271,14 @@ export function oauthFlow (cozy, storage, clientParams, onRegistered) {
       })
   }
 
-  return Promise.all([
-    storage.load(CredsKey),
-    storage.load(StateKey)
-  ])
+  const initialPromise = (ignoreCachedCredentials) ? storage.clear() : Promise.resolve()
+
+  return initialPromise.then(() => {
+    return Promise.all([
+      storage.load(CredsKey),
+      storage.load(StateKey)
+    ])
+  })
```

Comment:
I would have wrote it like that, for simplicity and to move this part out of the main promise chain:

```js
function oauthFlow (cozy, storage, clientParams, onRegistered, ignoreCachedCredentials = false) {
  if (ignoreCachedCredentials) {
    return storage.clear().then(() =>
      oauthFlow(cozy, storage, clientParams, onRegistered, false))
  }
  // ...
}
```